File: src/main/java/cn/huidu/device/api/sdk/data/program/area/content/DialClockNode.java
package cn.huidu.device.api.sdk.data.program.area.content;

import cn.huidu.device.api.sdk.data.program.other.Font;
import cn.huidu.device.api.sdk.data.program.other.ClockParam;
import cn.huidu.device.api.sdk.data.program.other.ClockParam.FmtStyle;
import cn.huidu.device.api.sdk.data.program.other.ClockParam.FormatColor;
import cn.huidu.device.api.sdk.data.program.other.ClockParam.StringColor;
import cn.huidu.device.api.sdk.data.program.other.ClockParam.Style;

public class DialClockNode extends ContentNode {
    private String timezone;
    private String timeOffset;
    private Font font;
    private StringColor title;
    private FormatColor date;
    private FormatColor week;
    private FormatColor time;
    private FormatColor ampm;
    private Style style;
    private int spacing;
    private boolean enabled12H;
    private boolean swapWeekAndDay;

    public DialClockNode() {
        this.setType("dialClock");
        title = new ClockParam().new StringColor();
        title.display = "0";
    }

    public DialClockNode(DialClockNode other) {
        this();
        this.timezone = other.timezone;
        this.timeOffset = other.timezone;
        this.font = other.font;
        this.title = other.title;
        this.date = other.date;
        this.week = other.week;
        this.time = other.time;
        this.ampm = other.ampm;
        this.style = other.style;
        this.enabled12H = other.enabled12H;
    }

    public String getTimezone() {
        return timezone;
    }

    public void setTimezone(String timezone) {
        this.timezone = timezone;
    }

    public String getTimeOffset() {
        return timeOffset;
    }

    public void setTimeOffset(String timeOffset) {
        this.timeOffset = timeOffset;
    }

    public Font getFont() {
        return font;
    }

    public void setFont(Font font) {
        this.font = font;
    }

    public StringColor getTitle() {
        return title;
    }

    public void setTitle(StringColor title) {
        this.title = title;
    }

    public FormatColor getDate() {
        return date;
    }

    public void setDate(FormatColor date) {
        this.date = date;
    }

    public FormatColor getWeek() {
        return week;
    }

    public void setWeek(FormatColor week) {
        this.week = week;
    }

    public FormatColor getTime() {
        return time;
    }

    public void setTime(FormatColor time) {
        this.time = time;
    }

    public Style getStyle() {
        return style;
    }

    public void setStyle(Style style) {
        this.style = style;
    }

    public FormatColor getAmpm() {
        return ampm;
    }

    public void setAmpm(FormatColor ampm) {
        this.ampm = ampm;
    }

    public Boolean getEnabled12H() {
        return enabled12H;
    }

    public void setEnabled12H(Boolean enabled12h) {
        enabled12H = enabled12h;
    }

    public FmtStyle getClockFmtStyle() {
        FmtStyle style = new ClockParam().new FmtStyle();
        if (swapWeekAndDay) {
            style.swap = "1";
        }
        if (spacing > 0) {
            style.space = Integer.toString(spacing);
        }

        return style;
    }

    public void setSwapWeekAndDay(Boolean swapWeekAndDay) {
        this.swapWeekAndDay = swapWeekAndDay;
    }

    public void setSpacing(int spacing) {
        this.spacing = spacing;
    }
}

File: python/sdk/data/area/DialClockNode.py
from typing import Optional
from sdk.data.area.ContentNode import ContentNode
from sdk.data.other.Font import Font
from sdk.data.other.ClockParam import ClockParam


class DialClockNode(ContentNode):
    def __init__(self, other: Optional['DialClockNode'] = None):
        super().__init__()
        self.set_type("dialClock")  # Set node type

        self.timezone: Optional[str] = None
        self.timeOffset: Optional[str] = None
        self.font: Optional[Font] = None
        self.title: ClockParam.StringColor = ClockParam.StringColor()
        self.title.display = "0"  # Set default display value
        self.date: Optional[ClockParam.FormatColor] = None
        self.week: Optional[ClockParam.FormatColor] = None
        self.time: Optional[ClockParam.FormatColor] = None
        self.ampm: Optional[ClockParam.FormatColor] = None
        self.style: Optional[ClockParam.Style] = None
        self.spacing: int = 0
        self.enabled12H: bool = False
        self.swapWeekAndDay: bool = False

        if other:
            self._copy_from(other)

    def _copy_from(self, other: 'DialClockNode'):
        super()._copy_from(other)  # Copy base class properties
        self.timezone = other.timezone
        self.timeOffset = other.timeOffset
        self.font = other.font
        self.title = other.title
        self.date = other.date
        self.week = other.week
        self.time = other.time
        self.ampm = other.ampm
        self.style = other.style
        self.spacing = other.spacing
        self.enabled12H = other.enabled12H
        self.swapWeekAndDay = other.swapWeekAndDay

    def get_timezone(self) -> Optional[str]:
        return self.timezone

    def set_timezone(self, timezone: str):
        self.timezone = timezone

    def get_time_offset(self) -> Optional[str]:
        return self.timeOffset

    def set_time_offset(self, time_offset: str):
        self.timeOffset = time_offset

    def get_font(self) -> Optional[Font]:
        return self.font

    def set_font(self, font: Font):
        self.font = font

    def get_title(self) -> ClockParam.StringColor:
        return self.title

    def set_title(self, title: ClockParam.StringColor):
        self.title = title

    def get_date(self) -> Optional[ClockParam.FormatColor]:
        return self.date

    def set_date(self, date: ClockParam.FormatColor):
        self.date = date

    def get_week(self) -> Optional[ClockParam.FormatColor]:
        return self.week

    def set_week(self, week: ClockParam.FormatColor):
        self.week = week

    def get_time(self) -> Optional[ClockParam.FormatColor]:
        return self.time

    def set_time(self, time: ClockParam.FormatColor):
        self.time = time

    def get_style(self) -> Optional[ClockParam.Style]:
        return self.style

    def set_style(self, style: ClockParam.Style):
        self.style = style

    def get_ampm(self) -> Optional[ClockParam.FormatColor]:
        return self.ampm

    def set_ampm(self, ampm: ClockParam.FormatColor):
        self.ampm = ampm

    def get_enabled12H(self) -> bool:
        return self.enabled12H

    def set_enabled12H(self, enabled: bool):
        self.enabled12H = enabled

    def get_swap_week_and_day(self) -> bool:
        return self.swapWeekAndDay

    def set_swap_week_and_day(self, swap: bool):
        self.swapWeekAndDay = swap

    def get_spacing(self) -> int:
        return self.spacing

    def set_spacing(self, spacing: int):
        self.spacing = spacing

    def get_clock_fmt_style(self) -> ClockParam.FmtStyle:
        fmt_style = ClockParam.FmtStyle()
        if self.swapWeekAndDay:
            fmt_style.swap = "1"
        if self.spacing > 0:
            fmt_style.space = str(self.spacing)
        return fmt_style

    def to_dict(self) -> dict:
        data = super().to_dict()
        data.update({
            "timezone": self.timezone,
            "timeOffset": self.timeOffset,
            "font": self.font.to_dict() if self.font else None,
            "title": self.title.__dict__ if self.title else None,
            "date": self.date.__dict__ if self.date else None,
            "week": self.week.__dict__ if self.week else None,
            "time": self.time.__dict__ if self.time else None,
            "ampm": self.ampm.__dict__ if self.ampm else None,
            "style": self.style.__dict__ if self.style else None,
            "spacing": self.spacing,
            "enabled12H": self.enabled12H,
            "swapWeekAndDay": self.swapWeekAndDay
        })
        return data

    def __repr__(self) -> str:
        return (f"DialClockNode(uuid={self.id}, name='{self.name}', "
                f"timezone='{self.timezone}', enabled12H={self.enabled12H})")